### PR TITLE
Feature: Delete From

### PR DESCRIPTION
**Pull request by dantengsky**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

WIP: Statement "delete from"

## Changelog

- New Feature
- Not for changelog (changelog entry is not required)

## Related Issues

Fixes #issue

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: query/src/storages/fuse/operations/delete_from.rs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                        } else {
+                            // Maybe there are rows should be deleted (false-positive of pruner)
+                            // use `delete_rows` to do the real data filtering.
+                            let new_segment = self.delete_rows(ctx.clone(), blocks, &expr).await?;
+                            segments_new.push(new_segment);
+                        }
+                    }
+
+                    let new_snapshot = self.merge_segments(&segments_kept, &segments_new);
+                    self.commit_deletion(new_snapshot).await?
```
Comment (sundy-li) [inline query/src/storages/fuse/operations/delete_from.rs:74]: So `deletion` will block `inserts` & `updates`?

Comment (dantengsky) [reply query/src/storages/fuse/operations/delete_from.rs:74]: `inserts` does not block other mutations. In this pr, a rather simple rule will be applied: when there are write-write conflicts,  first-committer-win.

Comment (sundy-li) [reply query/src/storages/fuse/operations/delete_from.rs:74]: So continuous small inserts will cause the heavy deletion retry ...

**Comment by dantengsky** [comment]
transfered to PR #5691

File: query/src/storages/fuse/operations/delete_from.rs (hunk 1 above)
Comment (dantengsky) [reply query/src/storages/fuse/operations/delete_from.rs:74]: > So continuous small inserts will cause the heavy deletion retry ...

er... too awkward... to catch this message months later. 

- yes, OCC implies the possibility of retrying
- deletion might be heavy, but the retrying of deletion might not be

**Pull request closed without merging**